extension/src/Simulation.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <godot_cpp/classes/object.hpp>
+#include <godot_cpp/core/class_db.hpp>
+#include <godot_cpp/variant/utility_functions.hpp>
+#include <vector>
+
+namespace OpenVic2 {
+	class Simulation : public godot::Object {
+		std::vector<size_t> exampleProvinces;
+
+		//BEGIN BOILERPLATE
```

Comment:
I'm not a fan of inheritance, but if we're gonna use a bunch of singletons, we should have a `Singleton` subclass of `godot::Object` that does most of this work. Just leaving `_bind_methods` to be implemented since that one changes a lot. Or a template, that works too.

Reply:
I agree that this would be the way to go.

I don't imagine that we would need much more than the core simulation to be a singleton since it can provide an interface that the Godot frontend uses

Reply:
Very cool.